src/main/java/org/truffleruby/core/regexp/RegexpCacheKey.java

```diff
@@ -19,19 +19,19 @@
     private final Rope rope;
     private final Encoding encoding;
     private final int options;
-    private final int hashCode;
+    private final Hashing hashing;
 
     public RegexpCacheKey(Rope rope, Encoding encoding, int options, Hashing hashing) {
         assert !(rope instanceof NativeRope);
         this.rope = rope;
         this.encoding = encoding;
         this.options = options;
-        this.hashCode = hashing.hash(rope.hashCode());
+        this.hashing = hashing;
     }
 
     @Override
     public int hashCode() {
-        return hashCode;
+        return hashing.hash(rope.hashCode());
```

Comment:
I can't say I like having a non-final hash code. But at least `Rope#hashCode` caches its calculation.

Reply:
Yeah, I think it doesn't matter much in practice. Hashing#hash(int) is just adding the seed to the cached Rope hash.